### ISSUE TITLE
As a user, I want to see a confirmation dialog when i try to delete a supporting document

### ISSUE THREAD

**Issue by jpye-finch** [opencrvs/opencrvs-core#4046]
### Description
- adds dialog if the user tries to delete supporting document
- from preview page (dark background one)
- from supporting docs page (red bin icon)

### Acceptance criteria

GIVEN I am on the supporting document
WHEN I click delete
THEN show confirmation model

GIVEN I am in document preview
WHEN I click delete
THEN show confirmation model

### Design
![image.png](https://images.zenhubusercontent.com/5e96badd8bea57051ce92eaf/fe9bb6d0-72fc-4653-8910-0f441e3c1c17)

### Dev tasks

- [ ] Task 1

**Comment by CornelMaze** [comment]
@jpye-finch So what happens on click of the confirmation delete button?

**Comment by jpye-finch** [comment]
we remove the supporting document

**Comment by anny320** [comment]
@CornelMaze please confirm if this is clear now

**Comment by CornelMaze** [comment]
@anny320 Yes this is clear. @Zangetsu101 already told me to hide the delete button which I did on #3617. So that ticket would be replaced with this, right?

**Comment by Zangetsu101** [comment]
@CornelMaze This confirmation modal would show up when trying to delete attachments from the documents section (in the form where we upload attachments) so the 2 tickets are kinda unrelated.

**Comment by euanmillar** [comment]
@jpye-finch suggest we remove the word "supporting" from the copy as the same component is used in certificate and signature upload.

**Comment by jpye-finch** [comment]
oh. Ideally I think it would be good to have specific copy for each case

**Comment by euanmillar** [comment]
Waiting for copy for all use cases

**Comment by jpye-finch** [comment]
@euanmillar i reviewed the different use case and documented in the ticket. We only need the model confirmation for supporting documents. The arn't needed for a few different reasons which I can discuss on a call

**Comment by Zangetsu101** [comment]
We have a hook called `useModal`. We should be able to use that to implement the modal in this ticket

**Comment by SyedaAfrida** [comment]
@Zangetsu101 @jpye-finch  can this ticket be closed?

**Comment by jpye-finch** [comment]
if it was built!?

**Comment by SyedaAfrida** [comment]
This is not built. Can you put this ticket under any Epic or make it a separate Epic @jpye-finch
